### PR TITLE
feat: calculate reward with getScaledPrice

### PR DESCRIPTION
**Pull request by chrisling-dev**

### CODE REVIEW

**Review by MeLlamoPablo — APPROVED** [review]
LGTM, pushed a nit

File: apps/frontend/core-dapp/src/stores/entities/TokenSenderEntity.ts [hunk 1]
```diff
@@ -66,30 +33,12 @@ export class TokenSenderEntity extends ContractStore<RootStore, SupportedContrac
     return this.call<GetPrice>('getPrice', [], { subscribe: false })
```
Comment (XavierEkkel) [inline apps/frontend/core-dapp/src/stores/entities/TokenSenderEntity.ts:33]: also dont need the price container anymore

File: apps/frontend/core-dapp/src/stores/entities/TokenSenderEntity.ts [hunk 2]
```diff
@@ -104,7 +53,7 @@ export class TokenSenderEntity extends ContractStore<RootStore, SupportedContrac
 
     const reward = feeInEth.mul(BigNumber.from(10).pow(ppoDecimals)).div(scaledPrice)
 
-    if (reward.gt(ppoBalance) && this.workaround_shouldCheckPpoBalance) {
+    if (reward.gt(ppoBalance)) {
       return BigNumber.from(0)
     }
```
Comment (XavierEkkel) [inline apps/frontend/core-dapp/src/stores/entities/TokenSenderEntity.ts:59]: also dont need `ppoPriceInEth` and `subscribeToPriceContainer` below

File: apps/frontend/core-dapp/src/stores/entities/TokenSenderEntity.ts [hunk 3]
```diff
@@ -6,54 +6,21 @@ import { RootStore } from '../RootStore'
...
 
-  constructor(
-    rootStore: RootStore,
-    address: string,
-    private workaround_overridePriceMultiplier?: BigNumber
-  ) {
+  constructor(rootStore: RootStore, address: string) {
     super(rootStore, 'DYNAMIC', TokenSenderAbi__factory as unknown as Factory)
     this.updateAddress(address)
     this.subscribeToPriceContainer()
```
Comment (XavierEkkel) [inline apps/frontend/core-dapp/src/stores/entities/TokenSenderEntity.ts:19]:
```suggestion
```

**Review by XavierEkkel — APPROVED** [review]
LGTM

**Pull request merged**